contracts/src/BackedCommunityTokenV1.sol

```diff
@@ -0,0 +1,169 @@
+import {IBackedCommunityTokenDescriptorV1} from "./interfaces/IBackedCommunityTokenDescriptorV1.sol";
+import {BackedCommunityStorageV1} from "./BackedCommunityStorageV1.sol";
+import {IERC721} from "../lib/openzeppelin-contracts/contracts/token/ERC721/IERC721.sol";
+import {ERC721SoulboundUpgradeable} from "./ERC721SoulboundUpgradeable.sol";
+
+contract BackedCommunityTokenV1 is
+    BackedCommunityStorageV1,
+    ERC721SoulboundUpgradeable,
+    IBackedCommunityTokenV1
+{
+    // ==== constructor ====
+    function initialize(address _descriptorAddress) public initializer {
+        __ERC721Soulbound_init("BackedCommunity", "BACKED");
+        descriptor = IBackedCommunityTokenDescriptorV1(_descriptorAddress);
+    }
+
+    // ==== state changing external functions ====
+
+    function addCategory(string memory displayName)
+        external
+        override
+        onlyOwner
+    {
+        categoryIdToDisplayName[totalCategoryCount++] = displayName;
+    }
+
+    function addSpecialAccessory(
+        IBackedCommunityTokenV1.Accessory memory accessory
+    ) external override onlyOwner {
+        accessoryIdToAccessory[
+            totalSpecialyAccessoryCount++
+        ] = IBackedCommunityTokenV1.Accessory({
+            name: accessory.name,
+            xpBased: accessory.xpBased,
+            artContract: accessory.artContract,
+            qualifyingXPScore: accessory.qualifyingXPScore,
+            xpCategory: accessory.xpCategory
+        });
+    }
+
+    function setCategoryScores(CategoryScoreChange[] memory changes)
+        external
+        override
+        onlyOwner
+    {
+        for (uint256 i = 0; i < changes.length; i++) {
+            _setCategoryScore(changes[i]);
+        }
+    }
+
+    function unlockAccessories(AccessoryUnlockChange[] memory changes)
+        external
+        override
+        onlyOwner
+    {
+        for (uint256 i = 0; i < changes.length; i++) {
+            _unlockAccessory(changes[i]);
+        }
+    }
+
+    function setEnabledAccessory(uint256 accessoryId) external override {
+        uint256 oldAccessoryId = addressToAccessoryEnabled[msg.sender];
+        IBackedCommunityTokenV1.Accessory
+            memory accessory = accessoryIdToAccessory[accessoryId];
+        if (accessory.xpBased) {
+            require(
+                addressToCategoryScore[msg.sender][accessory.xpCategory] >=
+                    accessory.qualifyingXPScore,
+                "BackedCommunityTokenV1: user does not qualify"
+            );
+        } else {
+            require(
+                addressToAccessoryUnlocked[msg.sender][accessoryId],
+                "BackedCommunityTokenV1: accessory not unlocked"
+            );
+        }
+        addressToAccessoryEnabled[msg.sender] = accessoryId;
+        emit AccessorySwapped(msg.sender, oldAccessoryId, accessoryId);
+    }
+
+    function linkBunnyPFP(uint256 tokenId) external override {
```

Comment:
If there isn't two way allowance then there needs to be some `revokeLinkedBunnyPFP(address)` that can only be called by the `bunnyPFPContractAddress` to delete the linked PFP when it's transferred.